### DISCUSSION TITLE
Problem with dual midi interface

### DISCUSSION THREAD

**Discussion by souravlal** [tttapa/Control-Surface#973] [Q&A]
Hi,
I have used your library with a Raspberry pi pico to build a midi controller having both Serial midi and usb midi interface. I have read the documentation and found that there is a scope for dual midi interface, so I used the sample code you have given. I can send midi messages to my pc and DAW software. But when I try to use serial midi interface by powering the raspberry pi pico with a 5V usb charger I can't send Serial midi messages, the raspberry pi pico only seems to work if it is connected to a pc(which has usb connectivity. When connected to the pc the serial midi messages are also sending along with usb midi messages without any problem). But when powered it with any other powers source, it doesn't send Serial messages. I have tried it with Arduino pro micro (native USB) it works fine.

**Comment by tttapa** [comment]
Please do not open multiple threads on the same topic.

The default behavior of the USB MIDI interface is to wait until the data can be sent (ignoring some limited buffering that takes place). If no USB connection is available, it will wait forever.

Changing that behavior would cause MIDI data to get lost, so I need to be cautious not to break other code. One possibility is to check if a USB connection is available, and drop outgoing messages as long as that's not the case. I would need a way to report such a failure back to the user, which is tricky to generalize for all errors, because the sending of MIDI packets happens asynchronously.

**Reply by souravlal** [reply]
Thanks for the reply.
But the same works without any problem on Arduino pro micro (Atmega32U4), the problem exists while using Raspberry pi pico. I didn't check any other USB boards.
